Lib/test/_test_multiprocessing.py

```diff
@@ -4472,17 +4473,19 @@ def check_semaphore_tracker_death(self, signum, should_die):
         # bpo-31310: if the semaphore tracker process has died, it should
         # be restarted implicitly.
         from multiprocessing.semaphore_tracker import _semaphore_tracker
-        _semaphore_tracker.ensure_running()
         pid = _semaphore_tracker._pid
+        if pid:
+            os.kill(pid, signal.SIGKILL)
+            time.sleep(0.5)  # give it time to die
+        with warnings.catch_warnings(record=True) as all_warn:
+            _semaphore_tracker.ensure_running()
+        pid = _semaphore_tracker._pid
+
         os.kill(pid, signum)
-        time.sleep(1.0)  # give it time to die
+        time.sleep(0.5)  # give it time to die
 
```

Comment:
Why do you reduce time?

Reply:
Because one second seems a lot to me for the signal to be delivered (checked in the buildbot) and this will make the test suite run a bit faster. I can undo this if we want to be conservative :).

Reply:
Please keep 1 second. While you saw the test passing, I'm sure that reducing the sleep will make the test failure more likely on some very busy buildbots.